datalad/tests/utils.py

```diff
@@ -231,6 +235,14 @@ def newfunc(*arg, **kw):
                 # well -- if something wrong just proceed with defaults
                 pass
 
+        # if DATALAD_TESTS_TEMPDIR is set, use that as directory,
+        # let mktemp handle it otherwise. However, an explicitly provided
+        # dir=... will override this.
+        directory = os.environ.get('DATALAD_TESTS_TEMPDIR')
+        if directory is not None and 'dir' not in tkwargs:
```

Comment:
mental note for myself:  apparently no need for explicit default for os.environ.get
note for Ben:  "if directory and ..." would be sufficient/more concise